src/main/java/uk/ac/ebi/eva/pipeline/io/readers/VcfHeaderReader.java

```diff
@@ -98,19 +98,33 @@ public VariantSourceEntity read() throws Exception {
             readAlreadyDone = true;
             return doRead();
         }
+
     }
 
-    private VariantSourceEntity doRead() {
+    private VariantSourceEntity doRead() throws IOException {
         if (variantReader == null) {
             throw new IllegalStateException("The method VcfHeaderReader.open() should be called before reading");
         }
         variantReader.pre();
         source.addMetadata(VARIANT_FILE_HEADER_KEY, variantReader.getHeader());
+
+        List<String> sampleNames = variantReader.getSampleNames();
+
+        Set<String> uniformSamples = new HashSet<String>();
+
+        for (String sample : sampleNames) {
+            if ((uniformSamples.contains(sample))) {
+                throw new IOException("Duplicate sample name given: " + sample);
```

Comment:
As you mention in the PR comments there was no suitable exception yet to cover this. But this one is too generic, because it could be thrown due to any issue with access to disk, etc. Maybe you could create a `DuplicateSamplesFoundException` inside this package (it wouldn't make sense to add it to `runner`)?